src/plugins/github/relationalView.test.js

```diff
@@ -390,6 +391,49 @@ describe("plugins/github/relationalView", () => {
     });
   });
 
+  describe("urlReferenceMap", () => {
+    it("should match example snapshot", () => {
+      // Given
+      const rv = exampleRelationalView();
+
+      // When
+      const map = rv.urlReferenceMap();
+
+      // Then
+      const addressStringsFor = (urls: string[]): mixed[] => {
+        const maybeString = (a) => (a ? NodeAddress.toString(a) : a);
+        return urls.map((url) => maybeString(map.get(url)));
+      };
+      expect(
+        addressStringsFor([
+          "https://github.com/decentralion",
+          "https://github.com/credbot",
+          "https://github.com/sourcecred-test/example-github",
+          "https://github.com/sourcecred-test/example-github/issues/1",
+          "https://github.com/sourcecred-test/example-github/pull/3",
+          "https://github.com/sourcecred-test/example-github/pull/5#pullrequestreview-100313899",
+          "https://github.com/sourcecred-test/example-github/issues/2#issuecomment-373768703",
+          "https://github.com/sourcecred-test/example-github/pull/3#issuecomment-369162222",
+          "https://github.com/sourcecred-test/example-github/pull/5#discussion_r171460198",
+          "https://github.com/sourcecred-test/example-github/commit/0a223346b4e6dec0127b1e6aa892c4ee0424b66a",
+        ])
+      ).toMatchInlineSnapshot(`
+        Array [
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"USERLIKE\\",\\"USER\\",\\"decentralion\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"USERLIKE\\",\\"BOT\\",\\"credbot\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"REPO\\",\\"sourcecred-test\\",\\"example-github\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"ISSUE\\",\\"sourcecred-test\\",\\"example-github\\",\\"1\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"PULL\\",\\"sourcecred-test\\",\\"example-github\\",\\"3\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"REVIEW\\",\\"sourcecred-test\\",\\"example-github\\",\\"5\\",\\"100313899\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"COMMENT\\",\\"ISSUE\\",\\"sourcecred-test\\",\\"example-github\\",\\"2\\",\\"373768703\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"COMMENT\\",\\"PULL\\",\\"sourcecred-test\\",\\"example-github\\",\\"3\\",\\"369162222\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"COMMENT\\",\\"REVIEW\\",\\"sourcecred-test\\",\\"example-github\\",\\"5\\",\\"100313899\\",\\"171460198\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"COMMIT\\",\\"MDY6Q29tbWl0MTIzMjU1MDA2OjBhMjIzMzQ2YjRlNmRlYzAxMjdiMWU2YWE4OTJjNGVlMDQyNGI2NmE=\\"]",
+        ]
+      `);
```

Comment:
@decentralion How is this as an alternative to the mega snapshot?

Reply:
This is basically what I had in mind. However, to make it easier to read, I recommend creating an array of {url, address} objects, and snapshotting that array. that way the reader can directly correlate the address to the url without needing to mentally track an index in both arrays.

Reply:
I tried a few options.

```js
type MapBased = Map<string, ?string>;
type ObjectBased = {url: string, address: ?string}[];
type TupleBased = [string, ?string][];
```

And found the tuple version most easy to read in the snapshot (how prettier stores it) and test error message context. All of them were either a lot larger than the inline snapshot, so I've moved that back to the snapshot file.